### PR TITLE
Add documentation on running EESSI from a virtual machine

### PR DESCRIPTION
**Pull request by laraPPr**
on Windows and MacOS

### CODE REVIEW

File: docs/getting_access/eessi_limactl.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Installing EESSI with limactl on MacOS
```
Comment (ocaisa) [inline docs/getting_access/eessi_limactl.md:1]:
```suggestion
# Installing EESSI with Lima on MacOS
```

File: docs/getting_access/eessi_limactl.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+## Installation of Lima
+
+see lima documentation: [https://lima-vm.io/docs/installation/](https://lima-vm.io/docs/installation/)
```
Comment (ocaisa) [inline docs/getting_access/eessi_limactl.md:5]:
```suggestion
See Lima documentation: [https://lima-vm.io/docs/installation/](https://lima-vm.io/docs/installation/)
```

File: docs/getting_access/eessi_limactl.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+``` { .bash .copy }
+brew install lima
+```
+
+## Installing EESSI in limactl with EESSI template
```
Comment (ocaisa) [inline docs/getting_access/eessi_limactl.md:11]:
```suggestion
## Installing EESSI in `limactl` with EESSI template
```

File: docs/getting_access/eessi_wsl.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+# create client configuration file for CernVM-FS (no squid proxy, 10GB local CernVM-FS client cache)
+sudo bash -c "echo 'CVMFS_CLIENT_PROFILE="single"' > /etc/cvmfs/default.local"
+sudo bash -c "echo 'CVMFS_QUOTA_LIMIT=10000' >> /etc/cvmfs/default.local"
+
+# make sure that EESSI CernVM-FS repository is accessible
+sudo cvmfs_config setup
+
+# start CernVM-FS on WSL
+sudo cvmfs_config wsl2_start
```
Comment (ocaisa) [inline docs/getting_access/eessi_wsl.md:92]: This should probably be separated out (or repeated) as it needs to run every time you start up the VM

**Review by ocaisa — CHANGES_REQUESTED** [review]

File: docs/getting_access/eessi_limactl.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  Found EESSI repo @ /cvmfs/software.eessi.io/versions/2023.06!
+  archdetect says x86_64/intel/haswell
+  Using x86_64/intel/haswell as software subdirectory.
+  Found Lmod configuration file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/.lmod/lmodrc.lua
+  Found Lmod SitePackage.lua file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/.lmod/SitePackage.lua
+  Using /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/modules/all as the directory to be added to MODULEPATH.
+  Using /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/intel/haswell/modules/all as the site extension directory to be added to MODULEPATH.
+  Initializing Lmod...
+  Prepending /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/modules/all to $MODULEPATH...
+  EPrepending site path /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/intel/haswell/modules/all to $MODULEPATH...
```
Comment (bedroge) [inline docs/getting_access/eessi_limactl.md:148]:
```suggestion
  Prepending site path /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/intel/haswell/modules/all to $MODULEPATH...
```

File: docs/getting_access/eessi_wsl.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  Found EESSI repo @ /cvmfs/software.eessi.io/versions/2023.06!
+  archdetect says x86_64/intel/haswell
+  Using x86_64/intel/haswell as software subdirectory.
+  Found Lmod configuration file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/.lmod/lmodrc.lua
+  Found Lmod SitePackage.lua file at /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/.lmod/SitePackage.lua
+  Using /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/modules/all as the directory to be added to MODULEPATH.
+  Using /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/intel/haswell/modules/all as the site extension directory to be added to MODULEPATH.
+  Initializing Lmod...
+  Prepending /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/intel/haswell/modules/all to $MODULEPATH...
+  EPrepending site path /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/intel/haswell/modules/all to $MODULEPATH...
```
Comment (bedroge) [inline docs/getting_access/eessi_wsl.md:108]:
```suggestion
  Prepending site path /cvmfs/software.eessi.io/host_injections/2023.06/software/linux/x86_64/intel/haswell/modules/all to $MODULEPATH...
```

File: mkdocs.yml [hunk 7]
```diff
@@ -26,6 +26,9 @@ nav:
     - Is EESSI already installed?: getting_access/is_eessi_accessible.md
     - Native: getting_access/native_installation.md
     - Container: getting_access/eessi_container.md
+    - Virtual machine on windows and macOS:
+      - Windows with WSL: getting_access/eessi_wsl.md
+      - MacOS with Limaclt: getting_access/eessi_limactl.md
```
Comment (bedroge) [inline mkdocs.yml:31]:
```suggestion
      - macOS with Lima: getting_access/eessi_limactl.md
```

File: docs/getting_access/eessi_limactl.md (hunk 1 above)
Comment (bedroge) [reply docs/getting_access/eessi_limactl.md:1]:
```suggestion
# Installing EESSI with Lima on macOS
```

File: docs/getting_access/eessi_wsl.md (hunk 4 above)
Comment (bedroge) [reply docs/getting_access/eessi_wsl.md:92]: I haven't tried it myself, but if I remember correctly, Kurt mentioned in the last monthly meeting that this was no longer required and you could just start autofs. Also see: https://klust.github.io/windows-client-HPC/4_Cluster_Stack/4_01_EESSI/#example-setup-on-fedora-remix-in-wsl2.

File: docs/getting_access/eessi_wsl.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+sudo bash -c "echo 'CVMFS_QUOTA_LIMIT=10000' >> /etc/cvmfs/default.local"
+
+# make sure that EESSI CernVM-FS repository is accessible
+sudo cvmfs_config setup
+
+# start CernVM-FS on WSL
+sudo cvmfs_config wsl2_start
+```
+
+EESSI should now be available in the Virtual machine
```
Comment (bedroge) [inline docs/getting_access/eessi_wsl.md:95]:
```suggestion
EESSI should now be available in the virtual machine:
```

File: docs/getting_access/eessi_limactl.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+        #!/bin/bash
+        set -eux -o pipefail
+        if ! timeout 30s bash -c "until ls /cvmfs/software.eessi.io >/dev/null 2>&1; do sleep 3; done"; then
+          echo >&2 "EESSI repository is not available yet"
+          exit 1
+        fi
+       hint: See "/var/log/cloud-init-output.log" in the guest
+    ```
+
+### Create the virtual machine with the `eessi.yaml` file:
```
Comment (bedroge) [inline docs/getting_access/eessi_limactl.md:122]:
```suggestion
### Create the virtual machine with the `eessi.yaml` file
```

File: docs/getting_access/eessi_wsl.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+Please create a default UNIX user account. The username does not need to match your Windows username.
+For more information visit: https://aka.ms/wslusers
+Enter new UNIX username: user
+New password:
+Retype new password:
+passwd: password updated successfully
+Installation successful!
+```
+
+## Installing EESSI in the Virtual machine
```
Comment (bedroge) [inline docs/getting_access/eessi_wsl.md:66]:
```suggestion
## Installing EESSI in the irtual machine
```

File: docs/getting_access/eessi_limactl.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+    - script: |
+        #!/bin/bash
+        set -eux -o pipefail
+        if ! timeout 30s bash -c "until ls /cvmfs/software.eessi.io >/dev/null 2>&1; do sleep 3; done"; then
+          echo >&2 "EESSI repository is not available yet"
+          exit 1
+        fi
+      hint: See "/var/log/cloud-init-output.log" in the guest
+    ```
+=== "Install a virtual machine with an Ubuntu image"
```
Comment (bedroge) [inline docs/getting_access/eessi_limactl.md:69]: I think I would just use one example here (either Debian or Ubuntu) to keep this a bit cleaner/shorter. Or can you make the file contents foldable/expandable?

Comment (bedroge) [reply docs/getting_access/eessi_limactl.md:69]: Alternatively we could store the templates in a git repo and provide links to them?

Comment (laraPPr) [reply docs/getting_access/eessi_limactl.md:69]: I have done it like on the native installation page, see https://www.eessi.io/docs/getting_access/native_installation/.
So only one shows up.

Comment (bedroge) [reply docs/getting_access/eessi_limactl.md:69]: Ah, okay, nice! :+1:

Comment (bedroge) [reply docs/getting_access/eessi_limactl.md:69]: In that case, I have another one (based on Rocky 9) 😄 
```
# A template to use the EESSI software stack (see https://eessi.io) on macOS
# $ limactl start ./eessi.yaml
# $ limactl shell eessi

images:
- location: "https://dl.rockylinux.org/pub/rocky/9.3/images/x86_64/Rocky-9-GenericCloud-Base-9.3-20231113.0.x86_64.qcow2"
  arch: "x86_64"
  digest: "sha256:7713278c37f29b0341b0a841ca3ec5c3724df86b4d97e7ee4a2a85def9b2e651"
- location: "https://dl.rockylinux.org/pub/rocky/9.3/images/aarch64/Rocky-9-GenericCloud-Base-9.3-20231113.0.aarch64.qcow2"
  arch: "aarch64"
  digest: "sha256:1948a5e00786dbf3230335339cf96491659e17444f5d00dabac0f095a7354cc1"
# Fallback to the latest release image.
# Hint: run `limactl prune` to invalidate the cache
- location: "https://dl.rockylinux.org/pub/rocky/9/images/x86_64/Rocky-9-GenericCloud.latest.x86_64.qcow2"
  arch: "x86_64"
- location: "https://dl.rockylinux.org/pub/rocky/9/images/aarch64/Rocky-9-GenericCloud.latest.aarch64.qcow2"
  arch: "aarch64"

mounts:
- location: "~"
- location: "/tmp/lima"
  writable: true
containerd:
  system: false
  user: false
provision:
- mode: system
  script: |
    #!/bin/bash
    sudo yum install -y https://ecsft.cern.ch/dist/cvmfs/cvmfs-release/cvmfs-release-latest.noarch.rpm
    sudo yum install -y cvmfs
    if [ ! -f /etc/cvmfs/default.local ]; then
        sudo echo "CVMFS_HTTP_PROXY=DIRECT" >> /etc/cvmfs/default.local
        sudo echo "CVMFS_QUOTA_LIMIT=10000" >> /etc/cvmfs/default.local
    fi
    sudo cvmfs_config setup
probes:
- script: |
    #!/bin/bash
    set -eux -o pipefail
    if ! timeout 30s bash -c "until ls /cvmfs/software.eessi.io >/dev/null 2>&1; do sleep 3; done"; then
      echo >&2 "EESSI repository is not available yet"
      exit 1
    fi
  hint: See "/var/log/cloud-init-output.log" in the guest
```

File: docs/getting_access/eessi_wsl.md (hunk 4 above)
Comment (laraPPr) [reply docs/getting_access/eessi_wsl.md:92]: Wanted to test the fedora remix does not seem to be free. But I can see if can do the same with one of the other wsl standard distributions.

Comment (laraPPr) [reply docs/getting_access/eessi_wsl.md:92]: And this should indeed be separate. but this only needs to be redone if the vm is terminated and than restarted. I will add this to the documentation.

Comment (laraPPr) [reply docs/getting_access/eessi_wsl.md:92]: I'll also look at the other thing maybe its better to go that route. I'll test it and than update the docs

File: docs/getting_access/eessi_limactl.md (hunk 11 above)
Comment (laraPPr) [reply docs/getting_access/eessi_limactl.md:69]: Done

File: docs/getting_access/eessi_wsl.md (hunk 4 above)
Comment (ocaisa) [reply docs/getting_access/eessi_wsl.md:92]: I think it is still needed with Ubuntu, but not with OpenSUSE

Comment (laraPPr) [reply docs/getting_access/eessi_wsl.md:92]: There is also an Ubuntu example on the page that Bob shared

Comment (laraPPr) [reply docs/getting_access/eessi_wsl.md:98]: I put it separately and add a link to how to set up the automounter. I've also tested it for `OpenSUSE-Tumbleweed` and `Ubuntu-24.04`. Which both were easy to do.

File: docs/getting_access/eessi_wsl.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+sudo bash -c "echo 'CVMFS_CLIENT_PROFILE="single"' > /etc/cvmfs/default.local"
+sudo bash -c "echo 'CVMFS_QUOTA_LIMIT=10000' >> /etc/cvmfs/default.local"
+
+# make sure that EESSI CernVM-FS repository is accessible
+sudo cvmfs_config setup
+```
+
+## Start cernVM-FS in Windows Subsystem for Linux
+
+When the virtual machine is restarted cernVM-FS needs to be remounted with following command.
```
Comment (ocaisa) [inline docs/getting_access/eessi_wsl.md:94]:
```suggestion
When the virtual machine is restarted CernVM-FS needs to be remounted with following command.
```

File: mkdocs.yml [hunk 13]
```diff
@@ -26,6 +26,9 @@ nav:
     - Is EESSI already installed?: getting_access/is_eessi_accessible.md
     - Native: getting_access/native_installation.md
     - Container: getting_access/eessi_container.md
+    - Virtual machine on windows and macOS:
```
Comment (ocaisa) [inline mkdocs.yml:29]:
```suggestion
    - Virtual machine on Windows and macOS:
```

**Review by ocaisa — CHANGES_REQUESTED** [review]

**Review by ocaisa — APPROVED** [review]

**Comment by ocaisa** [comment]
LGTM Thanks @laraPPr !

**Pull request merged**